ores/scoring_context.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import time
+from hashlib import sha1
+
+from revscoring import dependencies
+from revscoring.datasources import Datasource
+from revscoring.extractors import Extractor
+from revscoring.features import trim
+from revscoring.scorer_models import ScorerModel
+
+logger = logging.getLogger(__name__)
+
+
+class ScoringContext(dict):
+
+    def __init__(self, name, model_map, extractor):
+        """
+        :Parameters:
+            name : str
+               The  that this scorer is being used for
+            scorer_models : dict
+               A mapping between names and
+               :class:`revscoring.ScorerModel`
+               instances
+            extractor : :class:`revscoring.Extractor`
+               An extractor to use for gathering feature values
+        """
+        super().__init__()
+        self.name = str(name)
+        self.update(model_map)
```

Comment:
It's nice to reference the models contained by their names with the __getitem__, __setitem__ syntax.  We do this with contexts in the ScoringSystem too.  I'm not married to this pattern, but it's served us well so far.